Source/Fuse.Nodes/Visual.Children.LinkedList.uno

```diff
@@ -0,0 +1,216 @@
+using Uno;
+using Uno.Collections;
+using Uno.Text;
+using Uno.UX;
+
+using Fuse.Internal;
+
+namespace Fuse
+{
+	public partial class Node
+	{
+		internal Node Children_next;
+		internal Node Children_previous;
+		internal int Children_parentID = -1;
+	}
+
+	public partial class Visual
+	{
+		static int Children_thisIDEnumerator;
+		readonly int Children_thisID = Children_thisIDEnumerator++;
+
+		void Children_MakeParent(Visual parent, Node child)
+		{
+			if (child.Children_parentID != -1) throw new Exception();
+			child.Children_parentID = parent.Children_thisID;
+		}
+
+		void Children_MakeOrphan(Node child)
+		{
+			child.Children_parentID = -1;
+		}
+
+		internal Node Children_first;
+		internal Node Children_last;
+		int Children_count;
+
+		void Children_Add(Node n)
+		{
+			Children_CompleteCurrentAction();
+			Children_InvalidateCache();
+			Children_MakeParent(this, n);
+
+			if (Children_first == null) 
+			{
+				Children_first = n;
+				Children_last = n;
+			}
+			else
+			{
+				Children_last.Children_next = n;
+				n.Children_previous = Children_last;
+				Children_last = n;
+			}
+
+			Children_count++;
+		}
+
+		bool Children_Remove(Node n)
+		{
+			if (n.Children_parentID != Children_thisID) return false;
+
+			Children_CompleteCurrentAction();
+			Children_InvalidateCache();
+			Children_MakeOrphan(n);
+
+			if (Children_first == n)
+			{
+				Children_first = n.Children_next;
+				if (Children_last == n) Children_last = null;
+			}
+			else if (Children_last == n)
+			{
+				Children_last = n.Children_previous;
+			}
+			else
+			{
+				n.Children_previous.Children_next = n.Children_next;
+			}
+			n.Children_next = null;
+			n.Children_previous = null;
+			Children_count--;
+			return true;
+		}
+
+		int Children_IndexOf(Node n)
+		{
+			var k = Children_first;
+			int i = 0;
+			while (k != null)
+			{
+				if (k == n) return i;
+				i++;
+				k = k.Children_next;
+			}
+			return -1;
+		}
+
+		Node Children_ItemAt(int index)
+		{
+			Node k = Children_first;
+			int i = 0;
+			while (k != null)
+			{
+				if (i == index) return k;
+				i++;
+				k = k.Children_next;
+			}
+			throw new IndexOutOfRangeException();
+		}
+
+		// Returns the node appropriate to pass to InsertAt(node, elm)
+		// if the intention is to Insert(index, elm)
+		// This means it returns null if the index == 0, and throws exception
+		// if the index was outside bounds
+		Node Children_ItemBefore(int index)
+		{
+			if (index == 0) return null;
+			else return Children_ItemAt(index-1);
+		}
+
+		void Children_Insert(int index, Node newNode)
```

Comment:
It is still needed to implement `IList<T>`. Hopefully not used much in performance critical code paths, but must be there.